graphql/database/users/logSearch.js

```diff
@@ -1,18 +1,38 @@
 import moment from 'moment'
+import { nanoid } from 'nanoid'
 import UserModel from './UserModel'
 import UserSearchLogModel from './UserSearchLogModel'
-// import addVc from './addVc'
-// import checkSearchRateLimit from './checkSearchRateLimit'
 import { getTodaySearchCount } from './user-utils'
 import getUserSearchEngine from './getUserSearchEngine'
 
-/**
- * Log a user's search event and change related stats.
- * @param {object} userContext - The user authorizer object.
- * @param {string} userId - The user id.
- * @return {Promise<User>} A promise that resolves into a User instance.
- */
-const logSearch = async (userContext, userId, searchData = {}) => {
+const getSource = searchData => {
+  const validSearchSources = ['self', 'chrome', 'ff', 'tab']
+  return searchData.source && validSearchSources.indexOf(searchData.source) > -1
+    ? searchData.source
+    : null
+}
+
+const createUserSearchLogModel = async (
+  userContext,
+  userId,
+  causeId,
+  searchEngineId,
+  version,
+  isAnonymous,
+  source
+) => {
+  return UserSearchLogModel.create(userContext, {
+    userId,
+    timestamp: moment.utc().toISOString(),
+    ...(source && { source }),
+    searchEngine: searchEngineId || 'SearchForACause',
+    ...(causeId && { causeId }),
```

Comment:
This still needs some validation. See: https://github.com/gladly-team/tab/pull/1277#discussion_r907623912

plus some tests for logging null if the inputs don't pass validation.

Reply:
+ some tests for logging null if the inputs don't pass validation.